core/blockchain/impl/level_db_util.hpp

```diff
@@ -56,6 +63,17 @@ namespace kagome::blockchain {
       prefix::Prefix prefix, primitives::BlockNumber num,
       common::Hash256 block_hash, const common::Buffer &value);
 
+  /**
+   * Get an entry from the database
+   * @param db to get the entry from
+   * @param prefix, with which the entry was put into
+   * @param block_id - id of the block to get entry for
+   * @return encoded entry or error
+   */
+  outcome::result<common::Buffer> getWithPrefix(
```

Comment:
what if there are several entries with the same prefix? Shouldn't we return vector instead

Reply:
No, prefix is used to separate key space in a DB, like tables in relational DBs. You can check other methods dealing with prefixes, there were comments about it.